### ISSUE TITLE
Employee Permissions

### ISSUE THREAD

**Issue by odiea** [jekkos/opensourcepos#204]
Is there a way to grant an employee just the permission to add items in the pos screen or to prevent the Items from showing in the  header after permissions are granted? Currently an employee can not add an item unless they are given permission to  items .

**Comment by jekkos** [comment]
yes this is probably something I'd like to add to the application any time soon. Especially for the sales / receivings screen, some employees might only be able to create sales but not complete them, or suspend/unsuspend.. things like that.

for items it would make more sense to stick to the receivings module. Most of the time when we create new items it is in the scope of adding received orders to the system. If then an item is not existing, we create it on the fly and add it to the order at hand. Creating them directly in items module should be done without adding real quantities.

**Comment by odiea** [comment]
I do like it the way it is. It would be easier for some businesses not to have to add an item then go into receiving to add the qty to that item. That would appear to be additional work for some businesses.

**Comment by odiea** [comment]
Is there a way to build into add item that it automatically updates receiving?

**Comment by jekkos** [comment]
That is how it's always been, when adding an item and clicking submit, it should appear directly as a new order line in receivings screen. This functionality might be broken after the second button in that form was added

**Comment by odiea** [comment]
After I add a new item then go to details its listed under Inventory tracking control but if I go to reports receiving than nothing is listed as a received item.

**Comment by jekkos** [comment]
hmm normally when you do receivings, add a new item, previously it appeared in the order list, which had to be completed before that item would appear in the report for that receiving. Think this is the best way to add items datawise, all input is bundled in a receiving (and in inventory as well) which can be tracked more easily afterwards thanks to the receiving report

**Comment by odiea** [comment]
I guess I really never paid attention to adding items in the receiving module. I have always just added in the items module because that's where the barcode print feature is. I learn something new everyday. I will try adding items in receivings and see how that works out. Thanks

**Comment by jekkos** [comment]
One thing to keep in mind is that you should always create the item using the popup with quantity 0. The receiving then reflects adding the actual item to your inventory.

**Comment by sanjubaba1515** [comment]
If you want to make any change. 
PLEASE CONTACT ME AT SANJUBABA1515@GMAIL.COM, WHATS APP +919681971515

**Comment by daN4cat** [comment]
I checked the other day how to change permissions for Takings (delete & edit) and this needs to go down in the controller and table helper functions.
Also there is a graphical implication because when a checkbox is removed then the top arrow in the search bar should be removed.
Not a 5 min job I must say.

**Comment by gargmanishp** [comment]
We have just started using OSPOS. Few mods done by us are below. We need help on
1) Permissions - granular permissions - such as employees cannot see the whole customer list, but can create new customers or search customers
2) How do we create an API framework for OSPOS
3) Can we create multiple price lists according to location

Mods Done:
1) Delete buttons removed in takings, items, Takings update pop-up
2) Built a .NET API to pick weight of an item when quantity field is clicked - auto update total price once focus on quantity field goes away.
3) Customer search on takings screen works with name, email and name

**Comment by gargmanishp** [comment]
If we are guided by someone on permissions and API, we are willing to collaborate for joint development.
A lot of our work will be done only if multiple price lists can be managed.

**Comment by jekkos** [comment]
OK this sounds great. I'd like to see these improvements. Do you have them on github already? Send us a PR with the proposed changes so we can review/discuss and give feedback.

**Comment by jekkos** [comment]
Might be best to make pull requests per feature, so for things that are functionally related so integration is easier in the end

**Comment by daN4cat** [comment]
@gargmanishp any plan to contribute any of the suggested changes?

**Comment by gargmanishp** [comment]
We are now starting to work on 1) and 3) referenced in my post above dated Jan 11th. We have also created a new module where an employee can only view items and not edit them.

For 3) here is what we are planning to do:
Step 1: First we need to create the item location table where we will maintain the items location,  Cost price,  unit price based on item id.

Note: As per my understanding we need cost price and unit price because the cost price and unit price can be different for different item location.

Step 2: create the screen/form for inserting item location, cost price, unit price based on item id.

Step 3: Further we also need to show the items on the screen with their item location, cost price, unit price.

Step 4: On the item location screen, we need to give the edit button to edit the item location, cost price, unit price.

Step 5: On sale & receiving screen we need to modify the search to fetch the items based on the item location.

Step 6: while fetching the items based on item location, we also need to fetch the cost price and unit price from the item location table. (Because both prices can be different, so we can’t use the cost price and unit price from the ospos_items table )

**Comment by gargmanishp** [comment]
Just one comment. OSPOS already creates an item in a specific location if the item is created through the excel import. However if the item is created using the front end, then it is created across all locations. Hence, there is already a query which ties in an item to a location. However the price and cost info remains same across the locations. If these can be managed per location (and also Item Kits can be discounts/ multipliers on base products), then a lot of functionalities required by people including promotions etc can be managed through the KITS and Location prices.

**Comment by daN4cat** [comment]
Delete and permission addressed by #1539 

I'm not sure how much more we need of this. Moreover @gabrielperezs we never heard back from you, so not sure what you implemented or not.

**Comment by SteveIreland** [comment]
There are also some nice ideas here that need to be incorporated into the purchasing work #1479
